src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -8627,15 +8654,17 @@ void Unit::SetSpeedRate(UnitMoveType mtype, float rate)
     {
         // register forced speed changes for WorldSession::HandleForceSpeedChangeAck
         // and do it only for real sent packets and use run for run/mounted as client expected
-        ++ToPlayer()->m_forced_speed_changes[mtype];
+        ++m_forced_speed_changes[mtype];
 
         if (!IsInCombat())
             if (Pet* pet = ToPlayer()->GetPet())
                 pet->SetSpeedRate(mtype, m_speed_rate[mtype]);
     }
 
-    if (Player* playerMover = Unit::ToPlayer(GetUnitBeingMoved())) // unit controlled by a player.
+    if (IsMovedByClient()) // unit controlled by a player.
```

Comment:
This just sounds weird. I mean the naming "IsMovedByClient" + "controlled by player". So according naming, client is in control not player :)

Reply:
Context is key here. "MovedByClient" emphasis that the unit is not "MovedByServer". Also, if a (human) player controls a game client which controls a unit then by the transitive property of control, the human player controls the unit and the naming still works :)

Reply:
Also, even though it sounds a bit cheesy, you could choose to view the player as client of the server operator (i.e. Blizzard).